hasura.planx.uk/migrations/1699010412430_alter_table_public_flows_add_column_metabase_link/up.sql

```diff
@@ -0,0 +1,2 @@
+alter table "public"."flows" add column "metabase_link" text
+ null;
```

Comment:
nit: could this column name be generalised as `analytics_link`? Just thinking: since it's an external link, there's no reason councils couldn't set their own non-metabase thing here in the future? And while we use metabase now, we know the wider ODP metrics & analytics team is also exploring other platforms.

Reply:
Absolutely, that's a great point @jessicamcinchak!

Reply:
I think the followup task of allowing editors to manage this link via service settings is already be captured by [this roadmap ticket](https://miro.com/app/board/uXjVPrebzwQ=/?moveToWidget=3458764548469369468&cot=14 ) :+1:  Since a future phase, maybe no need to make a Trello ticket yet?

Reply:
Thanks for sharing that @jessicamcinchak, I had a look on Trello but I didn't think to check the roadmap.

Reply:
Fixed here: b4e3f1de31c9aa578bd4322f8c61083d797a13c4

Reply:
nit: since this migration hasn't run against `main` yet, you can actually simplify this into a single migration with the correct name and skip the `alter... rename...` ! 

It's a bit annoying mid-PR because you'll have to destroy/re-create Vultr for a fresh DB for it to run against, but it makes for a tidier version history

Reply:
Thanks for the comment @jessicamcinchak I should have added a note on this! 

I'd tried squashing the migration but it still started with the incorrect column name so it didn't really act as the tidy up I was hoping for. 

My plan is to do an interactive rebase to clean up the history to start off with `analytics_link` although when I'm reviewing I sometimes find it hard to see if feedback has been taken on if the commit history changes around. While I was waiting I thought I'd incrementally add commits as fixes then do a tidy up once they were approved.

Do you have any recommendations for better work flows to handle this sort of stuff?

Reply:
When I'm dealing with Hasura migrations on branches, I'd actually skip a Git workflow and just remove/edit the mgiration files directly. Eg, you can delete these two folders and their respective up/down sqls, create a new up migration to add the column name afresh and just make sure you've destroyed Vultr before you push your new commit. When merged, staging hasura won't ever know these first two iterations existed! 

Does that make sense?

Reply:
Hasura CLI also has some built-in commands for squashing migrations you could try: https://hasura.io/docs/latest/hasura-cli/commands/hasura_migrate_squash/

Reply:
That does make sense! Thanks @jessicamcinchak 😁

Reply:
I did have a go with squashing but I maybe didn't get the correct type of squash as it just combined the two i.e. start with the wrong name then rename in one `up.sql`.

Reply:
Yeah the Hasura CLI will always aim to preserve all commands I think, whereas in this case that's not actually necessary because none of these commands are on main yet!

Reply:
I've just pushed up a commit which removes the old migrations and adds the new ones. 

If I've understood correctly, as I deleted the Vultr instance when the CI runs it'll create a new Pizza environments with the files that are there now i.e. the old migrations in the commit history will have no impact. 

The new pizza environment will only know the files as they appear in 4ed65920ad1abee9d12f9bf9d467fccf1d160e06

When it comes to merging everything gets squashed anyway so there will effectively be no record of the `metabase_link`?

Reply:
Thanks for the guidance I was really keen to tidy up 🥳 I'm still getting my head around how to work with Hasura and the general vibe is it's too easy after working with Django/Rails migrations 😂